New erx

juggernautsei opened the pull request:
Here are the new completions I believe will pass inspection...

bradymiller commented on interface/weno/RxLog.php:
```diff
@@ -0,0 +1,58 @@
+require_once("../globals.php");
+
+?>
+<h1>Rx Log</h1>
```
Translate

bradymiller commented on interface/weno/RxLog.php:
```diff
@@ -0,0 +1,58 @@
+</tr>";
+while($row = sqlFetchArray($log)){
+
+	print "<tr><td>" .text($row['prescription_id'])."</td><td>".text($row['date']).
```
Good to start getting in habit of fomatting the date when outputting it. Try this instead:
``````
print "<tr><td>" .text($row['prescription_id'])."</td><td>".oeFormatShortDate(text($row['date'])).
``````
Note this function is always included whenever include globals and can see this function here:
https://github.com/openemr/openemr/blob/master/library/formatting.inc.php#L20
(as you'll note it will change it to the formatting that is set in Administraion->Globals->Locale)

bradymiller commented on interface/weno/adminClass.php:
```diff
@@ -0,0 +1,740 @@
+          
+          sqlStatement("ALTER TABLE `prescriptions` ADD `ntx` INT(2) NULL DEFAULT NULL AFTER `prn`, ADD `rtx` INT(2) NULL DEFAULT NULL AFTER `ntx`, ADD `txDate` DATE NOT NULL AFTER `rtx`");
+        
+          sqlStatement("ALTER TABLE `pharmacies` ADD `ncpdp` INT(12) NULL AFTER `email`, ADD `npi` INT(12) NULL AFTER `ncpdp`");
```
This is just 4 tables and see no great reason to go through all this to add them like this (note doing this makes it very difficult to update them). And very difficult to track changes to the other database tables as you are doing here. Strongly recommend placing these 4 tables(and the users, precsription, pharmacies table changes) in the database.sql and the upgrade sql script.

I do recommend bringing in the data to the tables here though.

juggernautsei replied:
So help me to understand setup because I will l have to in change the activation instruction.

Moving the alter tables to the upgrade SQL means that either I will need to add a script to the admin class file to do it for them because I am not thinking about tech activating this feature but people who don't know what they are doing at all. 

So, how is this going to work when the user gets to the admin to activate the feature. 

I suppose that the system could prompt the user to click a link to execute the sql_upgrade in a new window. Trying to nail down the logic flow to make this work seamlessly.

bradymiller replied:
First need to use standard practices for new 5.0.1 users and users upgrading to 5.0.1. Thus need to place these tables and tables mods in database.sql and the standard upgrade script for 5.0.0 to 5.0.1. 
I think what you are asking here is what to do for 5.0.0 users that want to use this feature. And in that case you can do something like what you are doing here (but that code would not go into the main codebase since it's not pertinent).

bradymiller commented on interface/weno/ajax_sample_code.php:
```diff
@@ -0,0 +1,149 @@
+
+  <div id="confirm">
+  <br><br>
+      <input type='submit' id='confirm_btn' value='<?php print xlt("Confirm") ?>' >
```
use xla since it within html attributes

bradymiller commented on interface/weno/ajax_sample_code.php:
```diff
@@ -0,0 +1,149 @@
+      <input type='submit' id='confirm_btn' value='<?php print xlt("Confirm") ?>' >
+  </div>
+   <div id="transmit">
+      <input type='submit' id='order' value='<?php print xlt("Transmit Order") ?>' >
```
use xla since it within html attributes

bradymiller commented on interface/weno/confirm.php:
```diff
@@ -0,0 +1,258 @@
+function validate(){
+	var pharmacy = document.getElementById('pharm');
+	if(text.value.length == 0){
+		alert('<?php print xlt("Must select a pharmacy first"); ?>');
```
use xls() since this within a js alert

bradymiller commented on interface/weno/confirm.php:
```diff
@@ -0,0 +1,258 @@
+
+  $drug = array(); //list of records that need to updated with pharmacy information
+while($list = sqlFetchArray($send)){
+	print "<tr align='center'><td>". xlt(text($list['drug'])) . " </td><td> " . xlt(text($list['quantity'])) . "</td></tr>";
```
Are you meaning to translate the drug name and quantity?

bradymiller replied:
Also shouldn't be translating drug names or quantity (I assume this is a number)

bradymiller commented on interface/weno/confirm.php:
```diff
@@ -0,0 +1,258 @@
+?>
+<div id="fields">
+<h3><?php echo xlt("Select Pharmacy"); ?></h3>
+	    Patient Default <br>
```
Translate

bradymiller commented on interface/weno/confirm.php:
```diff
@@ -0,0 +1,258 @@
+	    <?php if(!$patientPharmacy['name']){
+                   print "<b>".xlt("Please set pharmacy in patient\'s chart!")."</b><br> <br>";
+               }else{
+	    	      print xlt(text($patientPharmacy['name'])); 
```
do you mean to translate the pharmacy name?

juggernautsei replied:
I think I was being proactive and trying to get ahead of the review. So, anything that is going to be printed on the screen should be translated?

bradymiller replied:
Only translate stuff that should be translated. I do not think we should be translating names of pharmacies.

robertdown replied:
proper nouns shouldn't be translated

bradymiller commented on interface/weno/confirm.php:
```diff
@@ -0,0 +1,258 @@
+               }else{
+	    	      print xlt(text($patientPharmacy['name'])); 
+               } 
+	    	      ?><br> <br>
```
ONGOING REVIEW BOOKMARK

juggernautsei replied:
I think I was being proactive and trying to get ahead of the review. So, anything that is going to be printed on the screen should be translated?

bradymiller replied:
Only translate things that should be translated, not everything that is printed to screen,

bradymiller replied:
Another thing to note is the html escaping, which you are using correctly, and are used on everything that is printed to screen. Those are the text(), attr() functions. And note that the translation functions have that build in (xlt(), xla()). You've been doing this correctly in almost all of your code but the above xlt(text($patientPharmacy['name'])); the text() would be redundant since xlt already html esapes it. (of course, this should not be translated anyways)

sunsetsystems commented on interface/weno/admin.php:
```diff
@@ -0,0 +1,150 @@
+<!--
+<h3>Select State to Import</h3>
+<form method="post" action="pharmacyInsert.php">
+<select name="state">
```
There is a "States" list that you can use to generate this (see Administration -> Lists -> States).

sunsetsystems commented on interface/weno/adminClass.php:
```diff
@@ -0,0 +1,736 @@
+           sqlStatement($sqlPharmacy, $addPharmacy);
+*/
+
+          $sqlNarc =
```
In general it's not good to hard-code lists. What other options are available to you?

juggernautsei replied:
This list is in a table in the database. This is not hard coded.

sunsetsystems replied:
What I mean is the initialization of it is hard coded. Surely it's likely to become out of date between releases?

juggernautsei replied:
Rod, I apologize for being air head with what you are pointing out. Could you be a little more specific? Starting from the beginning hard code which list and which list do you think will be outdated? I am confused because the snippet above shows the $sqlNarc. So, I focused on that list.

sunsetsystems replied:
Yes I'm talking about the erx_narcotics table. You are loading it here with data embedded in this module. It looks like stuff that will change over time, so how will that be updated? If it will be reloaded from some other authoritative source then perhaps this module should instead be doing that.

sunsetsystems replied:
There's a similar question with drugPaidInsert.php where you are loading another table with a huge amount of embedded data.

juggernautsei replied:
Ok, that helps. This data will rarely change. I have been working with an anesthesiologist for the past 5 yrs and the list of narcotics that are not allowed to be electronically prescribed has not changed. However, this list was downloaded from the DEA site via PDF that was scrapped into this. The DEA only puts this list out in a PDF. So I have to manually check the site each year to see if anything has changed and in 5 years. It has not. Not to say it won't but change comes slowly. 
However, I am still open to a better way to do this.

sunsetsystems replied:
Thanks for the clarification, in that case I have no objection. Same thing with drugPaidInsert.php?

juggernautsei replied:
No, the drugPaidInsert has a method for updates. It gets updated once a month. I have tried the update method on the initial load and it does not go well at all. So I opted to do the initial load this way then the updates work.

bradymiller commented on interface/weno/confirm.php:
```diff
@@ -0,0 +1,258 @@
+	    	      ?><br> <br>
+	    	      
+        <?php print xlt("Mail Order") ?> <br>
+        <input type = 'radio' name = 'pharmacy' id = 'mailOrder' value = "<?php print attr($mailOrder['id']) ?>"><?php print xlt("CCS Medical 	14255 49th Street, North, Clearwater, FL 33762")."<br>" ?> 
```
Do not translate the mailing address.

bradymiller commented on interface/weno/formularyInsert.php:
```diff
@@ -0,0 +1,103 @@
+ 
+ //if(!empty($find['ndc'])){
+	 sqlStatement("INSERT INTO `erx_drug_paid` SET `drug_label_name` = ?, `NDC` = ?, `price_per_unit` = ? ", array($drugName,$ndc,$price));
+	 echo "Inserted ". $drugName ."<br> "; 	
```
text()

bradymiller replied:
above comment was not addressed

juggernautsei replied:
Addressed today

bradymiller commented on interface/weno/jsonScript.php:
```diff
@@ -0,0 +1,132 @@
+    if($sex == "Male"){ $sex = "M";}
+	if($sex == "Female"){$sex = "F";} //has to be a single letter
+	
+$patient ='[{"patient": {"lname" : "'.$plname.'","fname" : "'.$pfname.'","street" : "'.$pstreet.'","city" : "'.$pcity.'","postal" : "'.$ppostalCode.'","DOB" : "'.$DOB.'","Sex" : "'.$sex.'"}},';
```
Wouldn't it be safer to build the php array and then convert it to json via json_encode?(then don't need to worry about escaping values)

juggernautsei replied:
Safety was not a concern while trying to figure out how to build a very specific JSON array from all the gathered information then return that script to the page to be transmitted.  The data has to be in the order that it is configured. I couldn't see how to pull the data in the order that I have it with all of the headers and then convert it to a JSON format. I let the code developed as I figure things out. I did request input from you Brady but I understand you are bombarded day in and day out with development questions and you can't get to everyone. So, I did the best I could with what I know how to do and could find. 

Which values need escaping?

bradymiller replied:
Safety not really was a good term by me. For example, this will break if there is an apostrophe in a patient name. If you were to instead build $patient_array that is associative and looks like this:
$patient_array = array( "patient"=>array("lname"=>$plname,"fname"=>$pfname,...) );
And then convert it to json via:
$patient = json_encode($patient_array);

And if escaping issues do come up, then can leverage the options in the json_encode function to deal with it.

bradymiller replied:
Note that it will be a bit more complicated ion your case since you are adding the arrays together before making the json, so see below for how to do that.
So, for this function and others below would instead return the $patient_array.

bradymiller commented on interface/weno/jsonScript.php:
```diff
@@ -0,0 +1,132 @@
+
+function fillJsonProvider($fname,$lname,$npi,$wenoProvId,$name,$phone,$fax,$street,$city,$state,$postalCode,$wenoAccId,$wenoPass){
+	
+ $provider ='{"provider": {"provlname" : "'.$fname.'","provfname" : "'.$lname.'","provnpi" : '.$npi.',"facilityfax" : '.$fax.',"facilityphone" : '.$phone.',"facilityname" : "'.$name.'","facilitystreet" : "'.$street.'","facilitycity" : "'.$city.'","facilitystate" : "'.$state.'","facilityzip" : '.$postalCode.',"qualifier" : "'.$wenoProvId.'","wenoAccountId" : "'.$wenoAccId.'","wenoAccountPass" : "' ...
```
Wouldn't it be safer to build the php array and then convert it to json via json_encode?(then don't need to worry about escaping values)

bradymiller commented on interface/weno/jsonScript.php:
```diff
@@ -0,0 +1,132 @@
+
+function fillJsonPharmacy($storename,$storenpi,$pharmacy,$pharmacyPhone,$pharmacyFax){
+	
+$pharmacy ='{"pharmacy": {"storename" : "'.$storename.'","storenpi" : '.$storenpi.',"pharmacy" : '.$pharmacy.',"pharmacyPhone" : '.$pharmacyPhone.',"pharmacyFax" : '.$pharmacyFax.'}},';
```
Wouldn't it be safer to build the php array and then convert it to json via json_encode?(then don't need to worry about escaping values)

bradymiller commented on interface/weno/jsonScript.php:
```diff
@@ -0,0 +1,132 @@
+    $take         = $drugData['dosage'];
+
+function fillJsonScript($drugName,$drugNDC,$dateAdded,$quantity,$refills,$dateModified,$note,$take){
+$script ='{"script": {"drugName" : "'.$drugName.'","drug_NDC" : "'.$drugNDC.'","dateAdded" : "'.$dateAdded.'","quantity" : "'.$quantity.'","refills" : "'.$refills.'","dateModified" : "'.$dateModified.'","note" : "'.$note.'","take" : "'.$take.'"}}]';
```
Wouldn't it be safer to build the php array and then convert it to json via json_encode?(then don't need to worry about escaping values)

bradymiller commented on interface/weno/newSearchRxNorm.php:
```diff
@@ -0,0 +1,87 @@
+
+
+<form name="druglookup" method="post" action="">
+<input type="text" name="drug" size="15" value="<?php echo $drug; ?>">
```
text()

juggernautsei replied:
I am going to remove this file because it was an exploratory experiment.

bradymiller replied:
file hasn't been removed yet

juggernautsei replied:
removed this file

bradymiller commented on interface/weno/newSearchRxNorm.php:
```diff
@@ -0,0 +1,87 @@
+
+ini_set('display_errors', 1);
+ini_set('display_startup_errors', 1);
+error_reporting(E_ALL);
```
these 3 lines shoudn't be here

juggernautsei replied:
deleted file no longer a threat

bradymiller commented on interface/weno/newSearchRxNorm.php:
```diff
@@ -0,0 +1,87 @@
+
+<form name="druglookup" method="post" action="">
+<input type="text" name="drug" size="15" value="<?php echo $drug; ?>">
+<input type="submit" value="Submit">
```
translate

juggernautsei replied:
deleted file no longer a threat

bradymiller commented on interface/weno/newSearchRxNorm.php:
```diff
@@ -0,0 +1,87 @@
+
+if(!empty($_POST['drug'])){
+     $drug = $_POST['drug'];
+	 $xml = new SimpleXMLElement(file_get_contents("https://rxnav.nlm.nih.gov/REST/Prescribe/drugs?name=".$drug));
```
attr() around $drug

juggernautsei replied:
deleted file no longer a threat

bradymiller commented on interface/weno/newSearchRxNorm.php:
```diff
@@ -0,0 +1,87 @@
+echo "<pre>";
+//print_r($xml);
+echo "</pre>";
+echo xlt("Search term: "). text($xml->drugGroup->name) ."<br><br>";
```
No colon or space in translation constants.

juggernautsei replied:
deleted file no longer a threat

bradymiller commented on interface/weno/pharmacyInsert.php:
```diff
@@ -0,0 +1,76 @@
+	       $file = "./pharmacyList3.csv";
+	       break;
+
+	   defalut:
```
mispelled

bradymiller commented on interface/weno/transmitDataClass.php:
```diff
@@ -0,0 +1,123 @@
+	public function checkList($send){
+		$list = sqlFetchArray($send);
+		if(empty($list['id'])){
+			return xl("No Rx Selected ");
```
no trailing whitespace in translation constants

bradymiller replied:
above comment was not addressed

juggernautsei replied:
removed space

bradymiller commented on interface/weno/transmitDataClass.php:
```diff
@@ -0,0 +1,123 @@
+				$pFinfo = sqlQuery($sql, array($uid));
+			$wenoInfo = array();
+        			
+        $wsql = "SELECT gl_value FROM `globals` WHERE `gl_name` LIKE '%weno%'";
```
This makes me nervous. Why are you doing this?

juggernautsei replied:
being lazy in getting all the data from the globals.  I needed to get the id's and account info from the globals.

bradymiller replied:
need to use $GLOBALS instead :)

bradymiller commented on interface/weno/transmitDataClass.php:
```diff
@@ -0,0 +1,123 @@
+	}
+
+	public function active(){
+		$sql = "SELECT gl_value FROM `globals` WHERE `gl_name` LIKE 'weno_rx_enable'";
```
Why are you doing this when is already stored in $GLOBALS??

juggernautsei replied:
Not thinking I can just do $GLOBALS['weno_account_id'].

bradymiller replied:
Remove this function and instead simply use the actual Global rather than calling this function.

bradymiller commented on interface/weno/transmitDataClass.php:
```diff
@@ -0,0 +1,123 @@
+
+	public function validateWeno(){
+
+         $wenoInfo = "SELECT gl_value FROM globals WHERE gl_name LIKE ? ";
```
As above, makes me nervous. why are you doing this and why not just use the $GLOBALS settings?

bradymiller commented on interface/weno/validate.php:
```diff
@@ -0,0 +1,86 @@
+if(empty($weno[2]['gl_value'])){
+	print xlt("Weno Account Clinic ID information missing")."<br>";
+	exit;
+}
```
Why not just use $GLOBALS for above 3 if statements?

bradymiller commented on library/classes/Prescription.class.php:
```diff
@@ -479,11 +490,30 @@ function get_note() {
         $this->drug = $drug;
+		
+       if($GLOBALS['weno_rx_enable']){
+		$sql = "SELECT NDC FROM erx_drug_paid WHERE drug_label_name LIKE '%".$drug."%'";
```
use sql binding

bradymiller commented on templates/prescription/general_edit.html:
```diff
@@ -3,32 +3,54 @@
+    var x = document.forms["prescribe"]["dosage"].value;
+  var y = document.forms["prescribe"]["quantity"].value;
+    if (x == null || x == "") {
+        alert("Name must be filled out");
```
use xls()

bradymiller commented on templates/prescription/general_edit.html:
```diff
@@ -3,32 +3,54 @@
+        return false;
+    }
+  if (y == null || y == "") {
+        alert("<?php print xlt('Name must be filled out')?>");
```
use xls()

bradymiller commented on templates/prescription/general_edit.html:
```diff
@@ -3,32 +3,54 @@
+<script type="text/javascript" src="{$WEBROOT}/library/js/jquery-1.2.2.min.js"></script>
 <script type="text/javascript" src="{$WEBROOT}/library/js/jquery.bgiframe.min.js"></script>
 <script type="text/javascript" src="{$WEBROOT}/library/js/jquery.dimensions.pack.js"></script>
-<script type="text/javascript" src="{$WEBROOT}/library/js/jquery.autocomplete.pack.js"></script>
```
why remove this?

bradymiller commented on templates/prescription/general_edit.html:
```diff
@@ -3,32 +3,54 @@
-<script type="text/javascript" src="{$WEBROOT}/library/dialog.js?v={php} echo $v_js_includes; {/php}"></script>
-<script type="text/javascript" src="{php} echo $GLOBALS['assets_static_relative']; {/php}/jquery-min-1-2-2/index.js"></script>
+<script type="text/javascript" src="{$WEBROOT}/library/dialog.js"></script>
+<script type="text/javascript" src="{$WEBROOT}/library/js/jquery-1.2.2.min.js"></script>
```
revert the 4 changes above

bradymiller commented on templates/prescription/general_edit.html:
```diff
@@ -3,32 +3,54 @@
 {php}html_header_show();{/php}
 
 <link rel="stylesheet" href="{$CSS_HEADER}" type="text/css">
-<link rel="stylesheet" href="{$WEBROOT}/interface/themes/jquery.autocomplete.css" type="text/css">
```
why was this removed?

bradymiller commented:
Need to use standard practices for new 5.0.1 users and users upgrading to 5.0.1. Thus need to place the new sql tables and sql tables mods in database.sql and the standard upgrade script for 5.0.0 to 5.0.1.

bradymiller commented on interface/weno/jsonScript.php:
```diff
@@ -0,0 +1,132 @@
+$completeJson =  fillJsonPatient($pfname,$plname,$pstreet,$ppostalCode,$pcity,$DOB,$sex) . 
+                 fillJsonProvider($fname,$lname,$npi,$wenoProvId,$name,$phone,$fax,$street,$city,$state,$postalCode,$wenoAccId,$wenoPass) . 
+				 fillJsonPharmacy($storename,$storenpi,$pharmacy,$pharmacyPhone,$pharmacyFax) . 
+				 fillJsonScript($drugName,$drugNDC,$dateAdded,$quantity,$refills,$dateModified,$note,$take);
```
And then to continue from above, to build the json, would do:
``````
$complete_array = array_merge(fillJsonPatient($pfname,$plname,$pstreet,$ppostalCode,$pcity,$DOB,$sex), fillJsonProvider($fname,$lname,$npi,$wenoProvId,$name,$phone,$fax,$street,$city,$state,$postalCode,$wenoAccId,$wenoPass), ...);
``````
$complete_json = json_encode($complete_array);

bradymiller replied:
Please note I won't force this change for this code, but I highly suggest it, otherwise you will be tormented by escaping issues down the road when have funny characters in all the above fields (whereas if leverage json_encode, there are options in that function that can be used to simply escape every field if funny characters arise).

juggernautsei replied:
Brady, it seems we are miss communicating. array_merge is not going to produce the needed script transmit a prescription. Please review my notes in the  ajax_sample_code.php. The jsonscript.php file builds lines 70 -   113 programmatically. If it does not look like the hard code JSON script, it will be rejected by the WENO server.  So using the array_merge is not going to produce the script needed.

bradymiller replied:
But isn't your end goal here to simply produce the json (via the $completeJson echo below)?
Whether you create it a manually or convert it from a php array to json, then end result will still be json.

I am just suggesting that would be ideal to avoid manually creating the json. Am instead suggesting to instead leverage the conversion of php arrays to json via the json_encode() function in order to avoid character escaping pain down the road.(for example, place a apostrophe in a patient name and see what happens in your testing with current code)

juggernautsei replied:
No, the end goal is not to just produce JSON. It is to produce a specific JSON.

bradymiller replied:
I didn't realize there were different JSON formats. Is there a way you could send me a sample/dummy of what the specific JSON looks like in the output log?

bradymiller replied:
also, no need to waste to much of your time on this since this is just an optional recommendation. If your interested in the recommendation, though, I be happy to further investigate it.

juggernautsei replied:
Line 128 creates a log file of the JSON that was created and sent back to the browser via the ajax call. [View log file.](http://omp.openmedpractice.com/wenoInstall/interface/weno/log/22915jsonScript.txt)
Let me know if array_merge can produce this exactly.

bradymiller replied:
Hi @juggernautsei ,

Modified this script, which should produce the needed json:
https://github.com/bradymiller/openemr/commit/b322bc8272ef57ab6e9379a806a6c7013dfa3cc9

Easiest way for you to test it would be do simply copy the following file over:
https://github.com/bradymiller/openemr/blob/b322bc8272ef57ab6e9379a806a6c7013dfa3cc9/interface/weno/jsonScript.php

Let me know how it goes.
-brady

juggernautsei replied:
I just tested the code and it works. Thank you.
Lines 129-132 I thought would be good for logging like what they do in NewCrops but those lines have no bearing on the system functionality. It can be commented out or removed altogether.

bradymiller replied:
Oh, I just assumed it was temporary. If wish to store logs, then makes sense. Would store them in the sites/*/documents/weno_rx_log directory
(so basically just create the directory if it doesn't yet exist and store them there. Doing this keeps them separate on multisite, keeps them secure, makes it writable out of the box, and works with most users backup/restore strategies)

robertdown replied:
I have to agree with @bradymiller here, there are better ways to create the JSON.

```php
$patient = array(
    "lname" => $plname,
    "fname" => $pfname,
    ...
);

$provider = array("provlname" => $provlname, ...);

$output = array(
    "patient" => $patient,
    "provider" => $provider,
);

echo json_encode($output); // This gives you what you're manually creating
```

And if that needed to be wrapped in a for loop...

```php
$output = array()
foreach($fill as $item) {
   $output[] = createRow($item);
}
```

Then just encapsulate my first code output inside of createRow (Excluding the echoing json part) and you'll have what you need.

robertdown commented:
Also, this code introduces nearly 800 code standard errors into the codebase. I can fix nearly all of them automatically once this PR is ready to be integrated.

@juggernautsei what IDE are you using to code? I may be able to help get it setup to comply with the code standards

juggernautsei commented:
I am using sublime text

juggernautsei commented:
I'll start working on the SQL stuff tonight.

juggernautsei commented:
@robertdown 
Robert, can you help me out with what did you have in mind when you asked to separate the drugpaidinsert.php into its own SQL file? Do you want the file to be pure SQL?

juggernautsei commented:
Ready for review.

sjpadgett commented:
I'm really looking forward to this module. May I clone and test?

juggernautsei commented:
@sjpadgett 
please do clone and test. I will be doing the same after this last round of changes.  
http://open-emr.org/wiki/index.php/OpenEMR_ePrescribe#Activate_Service

This will help me very much. Thank you.

bradymiller commented on interface/weno/admin.php:
```diff
@@ -0,0 +1,97 @@
+<body class="body_top">
+<?php 
+
+if($active['gl_value'] != 1){ print xlt("You must activate Weno first!"); exit; } else {print xlt("Weno Service is Enabled")."<br>";} 
```
Isn't this just checking a global setting? If so, should just check the setting directly.

bradymiller commented on interface/weno/adminClass.php:
```diff
@@ -0,0 +1,161 @@
+          	);
+
+          sqlStatement(
+              "CREATE TABLE IF NOT EXISTS `erx_erx_narcotics` (
```
table name is wrong

juggernautsei replied:
corrected name

bradymiller commented on interface/weno/adminClass.php:
```diff
@@ -0,0 +1,161 @@
+                             PRIMARY KEY (`id`)
+                           ) ENGINE=InnoDB  DEFAULT CHARSET=utf8 AUTO_INCREMENT=570"
+          	);
+
```
Instead add these tables to the standard database sql file and the standard sql upgrade file. It's only 4 tables and makes way more sense (allows easy modification/improvement of tables if they are always there). And this will turn into a very important feature since plan to use for MU3.

juggernautsei replied:
On this one did you want me to add the tables to the database.sql or move them?

juggernautsei replied:
copied tables to the database.sql

bradymiller replied:
sorry on the delayed response. agree with copying tables to database.sql.

bradymiller replied:
And should remove it here since these tables will already be installed.

bradymiller commented on interface/weno/confirm.php:
```diff
@@ -0,0 +1,258 @@
+<div id="fields">
+<h3><?php echo xlt("Select Pharmacy"); ?></h3>
+	    <?php echo xlt("Patient Default"); ?> <br>
+	    <input type = 'radio' name = "pharmacy" id = 'patientPharmacy' value="<?php print $patientPharmacy['pharmacy_id'] ?>" checked="checked">
```
attr() around $patientPharmacy['pharmacy_id']

juggernautsei replied:
added

bradymiller commented on interface/weno/drugPaidInsert.php:
```diff
@@ -0,0 +1,13 @@
+<?php
```
need a standard header

juggernautsei replied:
added header

bradymiller commented on library/globals.inc.php:
```diff
@@ -2500,6 +2500,90 @@
+      xl('Enable Weno eRx Service'),
+      'bool',
+      '0',
+      xl('Enable Weno eRx Service.') + ' ' +
```
what is the stuff at the end of this line?

juggernautsei replied:
Don't know why it is there. Deleted.

juggernautsei replied:
removed and replaced with commas

bradymiller commented on sql/4_2_2-to-5_0_0_upgrade.sql:
```diff
@@ -127,6 +127,24 @@ ALTER TABLE `procedure_order_code` MODIFY `procedure_order_seq` int(11) NOT NULL
+-- 1. users
+--
+#IfMissingColumn users weno_prov_id
+ALTER TABLE `users` ADD `weno_prov_id` VARCHAR(15) NULL AFTER `newcrop_user_role`;
```
remove the AFTER stuff.

bradymiller commented on sql/4_2_2-to-5_0_0_upgrade.sql:
```diff
@@ -127,6 +127,24 @@ ALTER TABLE `procedure_order_code` MODIFY `procedure_order_seq` int(11) NOT NULL
+-- 2. prescriptions 
+--
+#IfMissingColumn prescriptions ntx 
+ALTER TABLE `prescriptions` ADD `ntx` INT(2) NULL DEFAULT NULL AFTER `prn`, ADD `rtx` INT(2) NULL DEFAULT NULL AFTER `ntx`, ADD `txDate` DATE NOT NULL AFTER `rtx`;
```
remove the after stuff
and make each add its own IfMissingColumn code block

bradymiller commented on sql/4_2_2-to-5_0_0_upgrade.sql:
```diff
@@ -127,6 +127,24 @@ ALTER TABLE `procedure_order_code` MODIFY `procedure_order_seq` int(11) NOT NULL
+-- 3. pharmacies 
+--
+#IfMissingColumn pharmacies ncpdp
+ALTER TABLE `pharmacies` ADD `ncpdp` INT(12) NULL AFTER `email`, ADD `npi` INT(12) NULL AFTER `ncpdp`;
```
remove the after stuff
and make each add its own IfMissingColumn code block

bradymiller commented on sql/4_2_2-to-5_0_0_upgrade.sql:
```diff
@@ -127,6 +127,24 @@ ALTER TABLE `procedure_order_code` MODIFY `procedure_order_seq` int(11) NOT NULL
+
 --
 -- Other tables do not need special treatment before conversion to InnoDB.
 -- Warning: running this query can take a long time.
```
Actually, this is the wrong upgrade file. upgrade changes should only be in below sql/5_0_0-to-5_0_1_upgrade.sql file

juggernautsei replied:
I put it in both because I didn't know for sure which to use. So, I'll remove the 4.2.2

bradymiller commented on sql/5_0_0-to-5_0_1_upgrade.sql:
```diff
@@ -129,6 +129,24 @@ VALUES ('page_validation','add_edit_event#theform_groups','/interface/main/calen
+-- 1. users
+--
+#IfMissingColumn users weno_prov_id
+ALTER TABLE `users` ADD `weno_prov_id` VARCHAR(15) NULL AFTER `newcrop_user_role`;
```
remove the AFTER stuff.

bradymiller commented on sql/5_0_0-to-5_0_1_upgrade.sql:
```diff
@@ -129,6 +129,24 @@ VALUES ('page_validation','add_edit_event#theform_groups','/interface/main/calen
+-- 2. prescriptions 
+--
+#IfMissingColumn prescriptions ntx 
+ALTER TABLE `prescriptions` ADD `ntx` INT(2) NULL DEFAULT NULL AFTER `prn`, ADD `rtx` INT(2) NULL DEFAULT NULL AFTER `ntx`, ADD `txDate` DATE NOT NULL AFTER `rtx`;
```
remove the after stuff
and make each add its own IfMissingColumn code block

bradymiller commented on sql/5_0_0-to-5_0_1_upgrade.sql:
```diff
@@ -129,6 +129,24 @@ VALUES ('page_validation','add_edit_event#theform_groups','/interface/main/calen
+-- 3. pharmacies 
+--
+#IfMissingColumn pharmacies ncpdp
+ALTER TABLE `pharmacies` ADD `ncpdp` INT(12) NULL AFTER `email`, ADD `npi` INT(12) NULL AFTER `ncpdp`;
```
remove the after stuff
and make each add its own IfMissingColumn code block

bradymiller commented on sql/5_0_0-to-5_0_1_upgrade.sql:
```diff
@@ -129,6 +129,24 @@ VALUES ('page_validation','add_edit_event#theform_groups','/interface/main/calen
+
 #IfNotTable therapy_groups
 CREATE TABLE `therapy_groups` (
   `group_id` int(11) NOT NULL auto_increment,
```
sql changes also need to be in sql/database.sql

juggernautsei replied:
I don't understand your comment.

bradymiller replied:
sql changes need to go into following:
1. sql/database.sql (for new installations)
2. sql/5_0_0-to-5_0_1_upgrade.sql (for upgraders)

So, these sql changes also need to be in sql/database.sql

bradymiller replied:
so for example, you would place the new ncpdp column here:
https://github.com/juggernautsei/openemr/blob/8dfe5d73a9b43e67fb7f7b5f38fdeaf0249b7cb2/sql/database.sql#L5387

bradymiller commented on templates/pharmacies/general_edit.html:
```diff
@@ -45,7 +45,18 @@
+	</td>
+</tr>
+<tr>
+	<td VALIGN="MIDDLE" >{xl t='NCPDP'}</td>
```
what is NCPDP?

juggernautsei replied:
The NCPDP Processor ID Number (BIN) is a six-digit number that health plans can use to process electronic pharmacy claims if they do not use pharmacy benefit cards with a magnetic stripe.
https://www.ncpdp.org/Resources/NCPDP-Processor-ID-(BIN)

juggernautsei replied:
nothing to do here

bradymiller commented on templates/prescription/general_edit.html:
```diff
@@ -4,21 +4,42 @@
+    var x = document.forms["prescribe"]["dosage"].value;
+  var y = document.forms["prescribe"]["quantity"].value;
+    if (x == null || x == "") {
+        alert("<?php print xls('Name must be filled out')?>");
```
Is this "Dosage" rather than "Name"?

bradymiller commented on templates/prescription/general_edit.html:
```diff
@@ -4,21 +4,42 @@
+        return false;
+    }
+  if (y == null || y == "") {
+        alert("<?php print xls('Name must be filled out')?>");
```
Is this "Quantity" rather than "Name"?

bradymiller commented on templates/prescription/general_list.html:
```diff
@@ -227,7 +268,7 @@
 <div class="text" style="margin-top:10px">{xl t='There are currently no prescriptions'}.</div>
 {/if}
-
+{php} echo $i; {/php}
```
what is this?

juggernautsei replied:
I don't see this in my copy of the program

juggernautsei replied:
removed

bradymiller commented on vendor/doctrine/cache/.travis.yml:
```diff
@@ -26,6 +26,7 @@ before_install:
     - if [[ $TRAVIS_PHP_VERSION = 7.* ]] ; then pecl config-set preferred_state beta; echo yes | pecl install apcu; fi;
     - if [[ $TRAVIS_PHP_VERSION != 'hhvm' ]]; then phpenv config-add ./tests/travis/php.ini; fi;
 
+
```
will be good to revert this. here's a quick way to revert a file like this which "sidesteps" git:
git checkout master
cp vendor/doctrine/cache/.travis.yml ../
git checkout new_erx
cp ../.travis.yml  vendor/doctrine/cache/.travis.yml
git commit -a -m "reverted the vendor/doctrine/cache/.travis.yml file to original"

juggernautsei replied:
I did not change this....?

bradymiller replied:
then who did ?
https://github.com/juggernautsei/openemr/blame/8dfe5d73a9b43e67fb7f7b5f38fdeaf0249b7cb2/vendor/doctrine/cache/.travis.yml#L28
:)

bradymiller commented:
review done. looking forward to next revision!
-brady

juggernautsei commented:
DB changes are ready for review.

juggernautsei commented:
What about me!  Should I rebase?
https://www.youtube.com/watch?v=teK7i7FGysA

bradymiller commented:
@juggernautsei ,
Wouldn't worry about rebasing this for now since it will easy to integrate with the most current codebase when it's ready to come into the codebase(this is because most of the code is very separated from the main codebase).
-brady

juggernautsei commented:
Well, now I'm in a bit of a pickle again. I did the rebase before your reply. But now I can't back out.
Tried looking it up with no help. The screen is stuck on new_erx|REBASE 1/18) even when it tells me that I am in the master. I tried git reset new_erx. No joy. 
 
![rebase](https://cloud.githubusercontent.com/assets/834958/26031965/49174198-3854-11e7-9993-1e73ba7ef1de.png)

bradymiller commented:
do following and show the output here:
git status

juggernautsei commented:
![rebase](https://cloud.githubusercontent.com/assets/834958/26032000/a401aea8-3855-11e7-922e-1acd03d81f1d.png)

bradymiller commented:
try:
git rebase --abort

juggernautsei commented:
seems so easy now. working properly now.

bradymiller commented:
nice! :)
do following again, just to make sure:
git status

juggernautsei commented:
![rebase](https://cloud.githubusercontent.com/assets/834958/26032060/c120cbde-3857-11e7-8873-e09788f7e707.png)

bradymiller commented:
Are those new files/dirs that you are working on?

juggernautsei commented:
No they are not

bradymiller commented:
I would probably do(note the dot after add):
git add .
git commit -a -m "new stuff maybe"
git log

bradymiller commented:
also, after show output of above, do this and show output:
git reflog

juggernautsei commented:
![rebase](https://cloud.githubusercontent.com/assets/834958/26032127/39af0c7c-3859-11e7-8c18-ece4441afee3.png)

bradymiller commented:
let me know what the following looks like:
git reflog

juggernautsei commented:
![rebase](https://cloud.githubusercontent.com/assets/834958/26032167/78f67ce8-385a-11e7-86a9-f33d3e82eebf.png)

bradymiller commented:
quite the mess :)

Do have a backup if things go crazy when you try to push it here, do the following:
git checkout origin/new_erx
(ignore the warning message)
git checkout -b backup_new_erx
(this will be a copy of your branch on github)

then go back to your new_erx and push it to github and cross fingers:
git checkout new_erx
git push origin new_erx

juggernautsei commented:
Thanks for working with me. It was 4 am and that was all that I could do. Sleep was heavy on me. I followed all the steps given. Now what would you like me to do next reflog

bradymiller commented:
Things look under control on the branch. Main question is if any of this code from this commit is meant to be here:
https://github.com/openemr/openemr/pull/605/commits/0b846c7b083d971759880735dadd613ba5e33c5b

juggernautsei commented:
Here are the real changes that I was submitting 
https://github.com/openemr/openemr/pull/605/commits/f933ec9f26f97e3a89e06cd747699a28426d72be

juggernautsei commented:
Review status please.

bradymiller commented:
@juggernautsei , I am still seeing quite a few of my comments that were not addressed on prior review (search for bradymiller at https://github.com/openemr/openemr/pull/605/files )

juggernautsei commented:
Apologies, I did not know understand to reply to every question. I will do that tonight.

juggernautsei commented:
@bradymiller  Kind Sir, https://github.com/openemr/openemr/pull/605/commits/48a027e988ec50b00bd088de8f01f92812b97580 was the last round of changes. Do you know when you will be able to take a look?

bradymiller commented on interface/usergroup/user_admin.php:
```diff
@@ -484,6 +484,9 @@ function authorized_clicked() {
 </td>
 </tr>
+<tr>
+<td><span class="text"><?php xl('Weno Provider ID','e'); ?>: </span></td><td><input type="text" name="erxprid" style="width:150px;"  value="<?php echo $iter["weno_prov_id"]?>"></td>
```
Since this is a new line of code, need to support html escaping. just copy this line instead:
``````
<td><span class="text"><?php echo xlt('Weno Provider ID'); ?>: </span></td><td><input type="text" name="erxprid" style="width:150px;"  value="<?php echo attr($iter["weno_prov_id"]); ?>"></td>
``````

bradymiller commented on interface/usergroup/usergroup_admin_add.php:
```diff
@@ -269,7 +269,9 @@ function authorized_clicked() {
 </tr>
-
+<tr>
+<td><span class="text"><?php xl('Weno Provider ID','e'); ?>: </span></td><td><input type="text" name="erxprid" style="width:120px;"  value="<?php echo $iter["weno_prov_id"]?>"></td>
```
Since this is a new line of code, need to support html escaping. just copy this line instead:
<td><span class="text"><?php echo xlt('Weno Provider ID'); ?>: </span></td><td><input type="text" name="erxprid" style="width:120px;"  value="<?php echo attr($iter["weno_prov_id"]); ?>"></td>

bradymiller commented on interface/weno/RxLog.php:
```diff
@@ -0,0 +1,58 @@
+ * @link    http://www.open-emr.org
+ */
+
+ $sanitize_all_escapes = true;		// SANITIZE ALL ESCAPES
```
remove flag (as of recently, the codebase no longer uses these flags :) )

bradymiller commented on interface/weno/RxLog.php:
```diff
@@ -0,0 +1,58 @@
+
+ $sanitize_all_escapes = true;		// SANITIZE ALL ESCAPES
+
+$fake_register_globals = false;		// STOP FAKE REGISTER GLOBALS
```
remove flag (as of recently, the codebase no longer uses these flags :) )

bradymiller commented on interface/weno/RxLog.php:
```diff
@@ -0,0 +1,58 @@
+<html>
+<head>
+<?php html_header_show();?>
+<link rel="stylesheet" href="<?php echo $css_header;?>" type="text/css">
```
Remove above line and use the following instead:
(this will basically bring in all the standard assets in addition to bootstrap/jquery in order to make it easier for developers):
``````
<?php require "{$GLOBALS['srcdir']}/templates/standard_header_template.php"; ?>

bradymiller commented on interface/weno/RxLog.php:
```diff
@@ -0,0 +1,58 @@
+</tr>";
+while($row = sqlFetchArray($log)){
+
+	print "<tr><td>" .text($row['prescription_id'])."</td><td>".oeFormatShortDate(text($row['date'])).
```
swap the text/oeFormatShortDate:
``````
text(oeFormatShortDate($row['date']))

bradymiller commented on interface/weno/admin.php:
```diff
@@ -0,0 +1,94 @@
+ * @link    http://www.open-emr.org
+ */
+
+$sanitize_all_escapes = true;		// SANITIZE ALL ESCAPES
```
remove flag (as of recently, the codebase no longer uses these flags :) )

bradymiller commented on interface/weno/admin.php:
```diff
@@ -0,0 +1,94 @@
+
+$sanitize_all_escapes = true;		// SANITIZE ALL ESCAPES
+
+$fake_register_globals = false;		// STOP FAKE REGISTER GLOBALS
```
remove flag (as of recently, the codebase no longer uses these flags :) )

bradymiller commented:
I am reviewing this now; note there are things in the current codebase that will be needed. so, we will need to do a rebase to current master branch. I can do this if needed (just takes me a couple minutes); just let me know and then I'd make another PR based on most recent codebase after I complete the code review.

bradymiller commented on interface/weno/admin.php:
```diff
@@ -0,0 +1,94 @@
+<head>
+     <title><?php print xlt("Weno Admin"); ?></title>
+    <?php html_header_show(); ?>
+    <link rel="stylesheet" href="<?php echo $css_header; ?>" type="text/css">
```
Remove above line and use the following instead:
(this will basically bring in all the standard assets in addition to bootstrap/jquery in order to make it easier for developers):
``````
<?php require "{$GLOBALS['srcdir']}/templates/standard_header_template.php"; ?>

bradymiller commented on interface/weno/admin.php:
```diff
@@ -0,0 +1,94 @@
+<body class="body_top">
+<?php 
+
+if($GLOBALS['weno_rx_enable'] != 1){ print xlt("You must activate Weno first!"); exit; } else {print xlt("Weno Service is Enabled")."<br>";} 
```
follow psr2 formatting for if/else:
http://www.php-fig.org/psr/psr-2/#if-elseif-else

bradymiller commented on interface/weno/admin.php:
```diff
@@ -0,0 +1,94 @@
+
+
+
+if(empty($exist[0]) && empty($exist[1]) && empty($exist[2]) && empty($exist[3])){
```
need 1 space after the if and before the {
``````
if (empty($exist[0]) && empty($exist[1]) && empty($exist[2]) && empty($exist[3])) {

bradymiller commented on interface/weno/admin.php:
```diff
@@ -0,0 +1,94 @@
+
+	$go = $tables->createTables();
+
+	print $go;
```
escape this
``````
print text($go);

bradymiller commented on interface/weno/admin.php:
```diff
@@ -0,0 +1,94 @@
+
+?>
+
+<h3>Select State to Import</h3>
```
translate:
``````
<h3><?php echo xlt('Select State to Import'); ?></h3>

bradymiller commented on interface/weno/admin.php:
```diff
@@ -0,0 +1,94 @@
+  
+   $drugData = $tables->drugTableInfo();
+if(!$drugData['ndc']){
+   echo "<button><a href='drugPaidInsert.php'>".xlt("Install Drugs Info")."</a></button> <br><br><br>".xlt("Be patient this may take a while");
```
note links and forms that open a new script have mechanism to prevent session leaks:
http://www.open-emr.org/wiki/index.php/OpenEMR_System_Architecture#PHP_Sessions_and_Browser_Windows

basically add onclick='top.restoreSession()' to this a tag:
``````
echo "<button><a href='drugPaidInsert.php' onclick='top.restoreSession()'>".xlt("Install Drugs Info")."</a></button> <br><br><br>".xlt("Be patient this may take a while");

bradymiller commented on interface/weno/admin.php:
```diff
@@ -0,0 +1,94 @@
+?>
+
+<h3>Select State to Import</h3>
+<form method="post" action="import_pharmacies.php" >
```
note links and forms that open a new script have mechanism to prevent session leaks:
http://www.open-emr.org/wiki/index.php/OpenEMR_System_Architecture#PHP_Sessions_and_Browser_Windows

basically add onsubmit='return top.restoreSession()' to this 'form' tag:
``````
<form method="post" action="import_pharmacies.php" onsubmit="return top.restoreSession()">

bradymiller commented on interface/weno/admin.php:
```diff
@@ -0,0 +1,94 @@
+echo generate_form_field(array('data_type'=>$GLOBALS['state_data_type'],'list_id'=>$GLOBALS['state_list'], 'field_id'=>'state'));
+
+?><br><br>
+<input type="submit" value="Import Pharmacies"><br>
```
need to translate:
``````
<input type="submit" value="<?php echo xla('Import Pharmacies'); ?>"><br>

bradymiller commented on interface/weno/admin.php:
```diff
@@ -0,0 +1,94 @@
+</form>
+<br><br>
+
+<?php  if(!empty($finish)){echo $finish . "with import";} ?>
```
Where is the $finish created. If not used, then remove it.

bradymiller commented on interface/weno/admin.php:
```diff
@@ -0,0 +1,94 @@
+
+?><br><br>
+<input type="submit" value="Import Pharmacies"><br>
+<p>Be patient, this can take a while.<br> There are 69852 records to sort through</p>
```
Above needs to be translated.
(and remove the second sentence)
(note there is a trick in INNODB to drastically speed up large imports; otherwise it takes a very long time)
(check out where ICD10/RXNORM/SNOMEd is imported for example)

bradymiller commented on interface/weno/adminClass.php:
```diff
@@ -0,0 +1,161 @@
+ * @link    http://www.open-emr.org
+ */
+
+class adminProperties {
```
a new class gets the happy class dance :)
![img_2681](https://cloud.githubusercontent.com/assets/278968/26423060/00ce752a-4081-11e7-80e9-47c1ba984734.GIF)

bradymiller commented on interface/weno/ajax_sample_code.php:
```diff
@@ -0,0 +1,149 @@
+ * @link    http://www.open-emr.org
+ */
+
+$sanitize_all_escapes = true;		// SANITIZE ALL ESCAPES
```
remove flag (as of recently, the codebase no longer uses these flags :) )

bradymiller commented on interface/weno/ajax_sample_code.php:
```diff
@@ -0,0 +1,149 @@
+
+$sanitize_all_escapes = true;		// SANITIZE ALL ESCAPES
+
+$fake_register_globals = false;		// STOP FAKE REGISTER GLOBALS
```
remove flag (as of recently, the codebase no longer uses these flags :) )

bradymiller commented on interface/weno/ajax_sample_code.php:
```diff
@@ -0,0 +1,149 @@
+<head>
+<title><?php print xlt("Weno Sample Code") ?></title>
+<link rel="stylesheet" href="<?php echo $css_header;?>" type="text/css">
+<link rel="stylesheet" href="<?php echo $GLOBALS['assets_static_relative']; ?>/jquery-ui-1-10-4/themes/sunny/jquery-ui.min.css" type="text/css" />
```
TODO for bradymiller
add support for sunny jquery-ui theme in at:
https://github.com/openemr/openemr/blob/master/library/templates/standard_header_template.php#L12-L13
TODO for bradymiller

bradymiller replied:
@juggernautsei ,
Before I do above, do you really need this sunny jquery-ui theme on this script?
I will bet that the out of box bootstrap will look just fine here.
-brady

bradymiller replied:
To keep this moving, I added the sunny theme to standard_header_template.php in main codebase.
So, now can add the following code(and remove lines 31, 32, and 49):
``````
<?php
html_header_show();
$include_standard_style_js = array("jquery-ui-sunny");
require "{$GLOBALS['srcdir']}/templates/standard_header_template.php";
?>

robertdown commented on interface/main/tabs/menu/menu_data.php:
```diff
@@ -117,6 +117,7 @@
       {"label":"Database","menu_id":"dba0","target":"dba","url":"/phpmyadmin/index.php","children":[],"requirement":0,"acl_req":["admin","database"],"global_req":"!disable_phpmyadmin_link"},
       {"label":"Certificates","menu_id":"adm0","target":"adm","url":"/interface/usergroup/ssl_certificates_admin.php","children":[],"requirement":0,"acl_req":["admin","users"]},
       {"label":"Native Data Loads","menu_id":"adm0","target":"adm","url":"/interface/super/load_codes.php","children":[],"requirement":0,"acl_req":["admin","super"]},
+      {"label":"Weno Management","menu_id":"adm0","target":"adm","url":"/interface/weno/admin.php","children":[],"requirement":0,"acl_req":["admin","super"]},
```
@bradymiller didn't I banish this file to an actual JSON file?

bradymiller replied:
@robertdown ,
This is just based on older code. After I finish the review, plan to rebase to most recent codebase.

bradymiller commented:
@robertdown @juggernautsei ,
Lets make the phpcs stuff optional for now. Sherwin likely does not even have composer installed. And this feature needs to get feature completed :)

bradymiller commented:
oops, closed the PR by mistake (just reopened it)

bradymiller commented on interface/weno/confirm.php:
```diff
@@ -0,0 +1,258 @@
+ * @link    http://www.open-emr.org
+ */
+ 
+$fake_register_globals=false;
```
remove flag (as of recently, the codebase no longer uses these flags :) )

bradymiller commented on interface/weno/confirm.php:
```diff
@@ -0,0 +1,258 @@
+ */
+ 
+$fake_register_globals=false;
+$sanitize_all_escapes=true;	
```
remove flag (as of recently, the codebase no longer uses these flags :) )

bradymiller commented on interface/weno/confirm.php:
```diff
@@ -0,0 +1,258 @@
+include_once('$srcdir/patient.inc');
+
+$date = date("Y-m-d");
+$pid = $GLOBALS['pid'];
```
This is odd since $pid should already be set.

bradymiller commented on interface/weno/confirm.php:
```diff
@@ -0,0 +1,258 @@
+<head>
+<?php html_header_show();?>
+<link rel="stylesheet" href="<?php echo $css_header;?>" type="text/css">
+<link rel="stylesheet" href="<?php echo $GLOBALS['assets_static_relative']; ?>/jquery-ui-1-10-4/themes/dot-luv/jquery-ui.min.css" type="text/css" />
```
What is with all the different jquery-ui themes?
Can this either be removed or changed to sunny??

bradymiller replied:
Remove lines 45, 46, and 123 . Then add following here(this will bring in the sunny jquery-ui theme):
<?php
$include_standard_style_js = array("jquery-ui-sunny");
require "{$GLOBALS['srcdir']}/templates/standard_header_template.php";
?>

bradymiller commented on interface/weno/drugPaidInsert.php:
```diff
@@ -0,0 +1,31 @@
+ */
+
+
+$sanitize_all_escapes = true;		// SANITIZE ALL ESCAPES
```
remove flag (as of recently, the codebase no longer uses these flags :) )

bradymiller commented on interface/weno/drugPaidInsert.php:
```diff
@@ -0,0 +1,31 @@
+
+$sanitize_all_escapes = true;		// SANITIZE ALL ESCAPES
+
+$fake_register_globals = false;		// STOP FAKE REGISTER GLOBALS
```
remove flag (as of recently, the codebase no longer uses these flags :) )

bradymiller commented on interface/weno/formularyInsert.php:
```diff
@@ -0,0 +1,103 @@
+ * @author  Sherwin Gaddis <sherwingaddis@gmail.com>
+ * @link    http://www.open-emr.org
+ */
+$fake_register_globals=false;
```
remove flag (as of recently, the codebase no longer uses these flags :) )

bradymiller commented on interface/weno/formularyInsert.php:
```diff
@@ -0,0 +1,103 @@
+ */
+$fake_register_globals=false;
+
+$sanitize_all_escapes=true;	
```
remove flag (as of recently, the codebase no longer uses these flags :) )

bradymiller commented on interface/weno/formularyInsert.php:
```diff
@@ -0,0 +1,103 @@
+
+ $file = "./Weno Exchange LLC/BSureFormulary.txt";
+ $lines = count(file($file));
+ echo $lines. "<br>";
```
escape:
``````
echo text($lines) . "<br>";

bradymiller commented on interface/weno/formularyInsert.php:
```diff
@@ -0,0 +1,103 @@
+ 
+ 
+ $i = 2;
+ do{
```
space
``````
do {

bradymiller commented on interface/weno/formularyInsert.php:
```diff
@@ -0,0 +1,103 @@
+ 
+ //if(!empty($find['ndc'])){
+	 sqlStatement("INSERT INTO `erx_drug_paid` SET `drug_label_name` = ?, `NDC` = ?, `price_per_unit` = ? ", array($drugName,$ndc,$price));
+	 echo text("Inserted "). $drugName ."<br> "; 	
```
should look like this:
``````
echo xlt("Inserted") . " " . text($drugName) ."<br> ";

bradymiller commented on interface/weno/formularyUpdate.php:
```diff
@@ -0,0 +1,101 @@
+ * @author  Sherwin Gaddis <sherwingaddis@gmail.com>
+ * @link    http://www.open-emr.org
+ */
+$fake_register_globals=false;
```
remove flag (as of recently, the codebase no longer uses these flags :) )

bradymiller commented on interface/weno/formularyUpdate.php:
```diff
@@ -0,0 +1,101 @@
+ * @link    http://www.open-emr.org
+ */
+$fake_register_globals=false;
+$sanitize_all_escapes=true;	
```
remove flag (as of recently, the codebase no longer uses these flags :) )

bradymiller commented on interface/weno/formularyUpdate.php:
```diff
@@ -0,0 +1,101 @@
+
+ $file = "./Weno Exchange LLC/BSureFormulary.txt";
+ $lines = count(file($file));
+ echo $lines. "<br>";
```
escape:
``````
echo text($lines) . "<br>";

bradymiller commented on interface/weno/formularyUpdate.php:
```diff
@@ -0,0 +1,101 @@
+ 
+ 
+ $i = 2;
+ do{
```
space
``````
do {

bradymiller commented on interface/weno/formularyUpdate.php:
```diff
@@ -0,0 +1,101 @@
+ do{
+ $fileName = new SplFileObject($file);
+ $fileName->seek($i);
+ echo $fileName."<br>";   //return the first line from file
```
escape:
``````
echo text($fileName) . "<br>";   //return the first line from file

bradymiller commented on interface/weno/formularyUpdate.php:
```diff
@@ -0,0 +1,101 @@
+ 
+ echo $in[2]."<br>";
+ echo $in[5]."<br>";
+ echo $in[8]."<br>";
```
wrap text() around the variables in 3 lines above

bradymiller commented on interface/weno/import_pharmacies.php:
```diff
@@ -0,0 +1,106 @@
+<?php
```
For the above interface/weno/img/progress.gif file, better to place this at public/images/

bradymiller commented on interface/weno/import_pharmacies.php:
```diff
@@ -0,0 +1,106 @@
+ * @link    http://www.open-emr.org
+ */
+
+$sanitize_all_escapes = true;   // SANITIZE ALL ESCAPES
```
remove flag (as of recently, the codebase no longer uses these flags :) )

bradymiller commented on interface/weno/import_pharmacies.php:
```diff
@@ -0,0 +1,106 @@
+
+$sanitize_all_escapes = true;   // SANITIZE ALL ESCAPES
+
+$fake_register_globals = false;   // STOP FAKE REGISTER GLOBALS
```
remove flag (as of recently, the codebase no longer uses these flags :) )

bradymiller commented on interface/weno/jsonScript.php:
```diff
@@ -0,0 +1,132 @@
+ * @author  Brady Miller <brady.g.miller@gmail.com>
+ * @link    http://www.open-emr.org
+ */
+
```
ONGOING REVIEW PLACEMARKER

bradymiller commented on interface/weno/jsonScript.php:
```diff
@@ -0,0 +1,132 @@
+ * @link    http://www.open-emr.org
+ */
+
+$fake_register_globals=false;
```
remove flag (as of recently, the codebase no longer uses these flags :) )

bradymiller commented on interface/weno/jsonScript.php:
```diff
@@ -0,0 +1,132 @@
+ */
+
+$fake_register_globals=false;
+$sanitize_all_escapes=true;
```
remove flag (as of recently, the codebase no longer uses these flags :) )

bradymiller commented on interface/weno/jsonScript.php:
```diff
@@ -0,0 +1,132 @@
+include_once("transmitDataClass.php"); 
+
+$date = date("Y-m-d");
+$pid = $GLOBALS['pid'];
```
this is odd since $pid should already be set

bradymiller commented on interface/weno/markTx.php:
```diff
@@ -0,0 +1,67 @@
+ * @link    http://www.open-emr.org
+ */
+ 
+$fake_register_globals=false;
```
remove flag (as of recently, the codebase no longer uses these flags :) )

bradymiller commented on interface/weno/markTx.php:
```diff
@@ -0,0 +1,67 @@
+ 
+$fake_register_globals=false;
+
+$sanitize_all_escapes=true;	
```
remove flag (as of recently, the codebase no longer uses these flags :) )

bradymiller commented on interface/weno/markTx.php:
```diff
@@ -0,0 +1,67 @@
+	//See if the value is set 
+	$check = "SELECT ntx FROM prescriptions WHERE id = ?";
+	$getVal = sqlStatement($check, array($e_script[1]));
+	$val = sqlFetchArray($getVal);
```
more straightforward code for above 2 lines would be:
``````
$val = sqlQuery($check, array($e_script[1]));

bradymiller commented on interface/weno/markTx.php:
```diff
@@ -0,0 +1,67 @@
+
+$e_script = explode("-", $script);
+
+if($e_script[0] === "NewRx"){
```
spacing:
``````
if ($e_script[0] === "NewRx") {

bradymiller commented on interface/weno/markTx.php:
```diff
@@ -0,0 +1,67 @@
+	
+	//If the value is not set to 1 then set it for new rx to transmit 
+	// ToDo add transmit date
+	if(empty($val['ntx'])){	
```
spacing

bradymiller commented on interface/weno/markTx.php:
```diff
@@ -0,0 +1,67 @@
+
+}
+ //There is a flaw in the logic because if the doc goes back in to edit the script this date gets wiped out.
+if($e_script[0] === "RefillRx"){
```
spacing

bradymiller commented on interface/weno/markTx.php:
```diff
@@ -0,0 +1,67 @@
+	
+}
+
+if($_GET['arr']){
```
spacing

bradymiller commented on interface/weno/markTx.php:
```diff
@@ -0,0 +1,67 @@
+	$i = count($scriptUpdate) - 1; //Since first number is always the pharmacy -1 
+	$ii = 1;
+	 
+    while($i >= $ii){
```
spacing

bradymiller commented on interface/weno/pharmacyInsert.php:
```diff
@@ -0,0 +1,76 @@
+ * @author  Sherwin Gaddis <sherwingaddis@gmail.com>
+ * @link    http://www.open-emr.org
+ */
+$fake_register_globals=false;
```
remove flag (as of recently, the codebase no longer uses these flags :) )

bradymiller commented on interface/weno/pharmacyInsert.php:
```diff
@@ -0,0 +1,76 @@
+ * @link    http://www.open-emr.org
+ */
+$fake_register_globals=false;
+$sanitize_all_escapes=true;
```
remove flag (as of recently, the codebase no longer uses these flags :) )

bradymiller commented on interface/weno/pharmacyInsert.php:
```diff
@@ -0,0 +1,76 @@
+ require_once("../globals.php");
+
+
+$state = filter_input(INPUT_POST, state);
```
doesn't state in this function need quotes around it?

juggernautsei replied:
removing this file, better feature built import_pharmcies.php

bradymiller commented on interface/weno/pharmacyInsert.php:
```diff
@@ -0,0 +1,76 @@
+
+
+$state = filter_input(INPUT_POST, state);
+echo $state;
```
escape:
``````
echo text($state);

juggernautsei replied:
done

bradymiller commented on interface/weno/pharmacyInsert.php:
```diff
@@ -0,0 +1,76 @@
+$state = filter_input(INPUT_POST, state);
+echo $state;
+
+switch($state){
```
spacing

juggernautsei replied:
removed

bradymiller commented on interface/weno/pharmacyInsert.php:
```diff
@@ -0,0 +1,76 @@
+
+switch($state){
+	case (preg_match("/^[A-Ka-k]/i", $state) ? true : false) :
+	       echo "State is between A - K";
```
is this debugging stuff?

juggernautsei replied:
removed

bradymiller commented on interface/weno/pharmacyInsert.php:
```diff
@@ -0,0 +1,76 @@
+	       break;
+
+	case (preg_match("/^[L-Nl-n]/i", $state) ? true : false) :
+	       echo "State is between L - N";
```
is this debugging stuff?

juggernautsei replied:
removed

bradymiller commented on interface/weno/pharmacyInsert.php:
```diff
@@ -0,0 +1,76 @@
+	       break;
+
+	case (preg_match("/^[O-Zo-z]/i", $state) ? true : false) :
+	       echo "State is between O - Z";
```
is this debugging stuff?

juggernautsei replied:
removed

bradymiller commented on interface/weno/pharmacyInsert.php:
```diff
@@ -0,0 +1,76 @@
+} 
+
+ $lines = count(file($file));
+ echo $lines. "<br>";
```
escape:
``````
echo text($lines) . "<br>";

juggernautsei replied:
removed

bradymiller commented on interface/weno/pharmacyInsert.php:
```diff
@@ -0,0 +1,76 @@
+ 
+ $i = 0;
+
+while($i <= $lines){
```
spacing

juggernautsei replied:
removed

bradymiller commented on interface/weno/pharmacyInsert.php:
```diff
@@ -0,0 +1,76 @@
+ $lineParts = explode(",", $lineToInsert);	
+ $i++;
+ 
+if($state !== $lineParts[7]){ continue; }
```
spacing

juggernautsei replied:
removed

bradymiller commented on interface/weno/transmit.inc.php:
```diff
@@ -0,0 +1,55 @@
+ * @link    http://www.open-emr.org
+ **/
+
+class transmitData {
```
a new class gets the happy class dance :)
![img_2681](https://cloud.githubusercontent.com/assets/278968/26423060/00ce752a-4081-11e7-80e9-47c1ba984734.GIF)

bradymiller commented on interface/weno/transmitDataClass.php:
```diff
@@ -0,0 +1,99 @@
+ * @link    http://www.open-emr.org
+ **/
+
+class transmitData {
```
Why do you have 2 classes of the same name?

juggernautsei replied:
Easy, I started with the transmit.inc.php and realized I was building a class and changed it's name. I have removed the include file.

bradymiller commented on interface/weno/validate.php:
```diff
@@ -0,0 +1,84 @@
+ * @link    http://www.open-emr.org
+ **/
+
+$fake_register_globals=false;
```
remove flag (as of recently, the codebase no longer uses these flags :) )

juggernautsei replied:
done

bradymiller commented on interface/weno/validate.php:
```diff
@@ -0,0 +1,84 @@
+ **/
+
+$fake_register_globals=false;
+$sanitize_all_escapes=true;	
```
remove flag (as of recently, the codebase no longer uses these flags :) )

juggernautsei replied:
done

bradymiller commented on interface/weno/validate.php:
```diff
@@ -0,0 +1,84 @@
+include_once('../globals.php');
+include_once('transmitDataClass.php');
+
+$pid = $GLOBALS['pid'];
```
this is odd since $pid should already be set

juggernautsei replied:
done

bradymiller commented on interface/weno/validate.php:
```diff
@@ -0,0 +1,84 @@
+$patient = $validation->validatePatient($pid);
+$pharmacy = $validation->patientPharmacyInfo($pid);
+
+if(empty($GLOBALS['weno_account_id'])){
```
spacing

bradymiller commented on interface/weno/validate.php:
```diff
@@ -0,0 +1,84 @@
+
+
+
+if(empty($GLOBALS['weno_provider_id'])){
```
spacing

bradymiller commented on interface/weno/validate.php:
```diff
@@ -0,0 +1,84 @@
+	exit;
+}
+
+if(empty($patient['DOB'])){
```
spacing

bradymiller commented on interface/weno/validate.php:
```diff
@@ -0,0 +1,84 @@
+	exit;
+}
+
+if(empty($patient['street'])){
```
spacing

bradymiller commented on interface/weno/validate.php:
```diff
@@ -0,0 +1,84 @@
+	exit;
+}
+
+if(empty($patient['postal_code'])){
```
spacing

juggernautsei replied:
removed all spacing

bradymiller commented on interface/weno/validate.php:
```diff
@@ -0,0 +1,84 @@
+	exit;
+}
+
+if(empty($patient['city'])){
```
spacing

bradymiller commented on interface/weno/validate.php:
```diff
@@ -0,0 +1,84 @@
+	exit;
+}
+
+if(empty($patient['state'])){
```
spacing

bradymiller commented on interface/weno/validate.php:
```diff
@@ -0,0 +1,84 @@
+	exit;
+}
+
+if(empty($patient['sex'])){
```
spacing

bradymiller commented on interface/weno/validate.php:
```diff
@@ -0,0 +1,84 @@
+	exit;
+}
+
+if(empty($pharmacy['name'])){
```
spacing

bradymiller commented on library/ajax/drug_autocomplete/index.php:
```diff
@@ -0,0 +1,28 @@
+<!doctype html>
```
needs license/copyright header

juggernautsei replied:
removed file

bradymiller commented on library/ajax/drug_autocomplete/index.php:
```diff
@@ -0,0 +1,28 @@
+<html lang="en">
+<head>
+  <meta charset="utf-8">
+  <title>Demo</title>
```
translate (use xlt)

bradymiller commented on library/ajax/drug_autocomplete/index.php:
```diff
@@ -0,0 +1,28 @@
+<body> 
+
+	<form action='' method='post'>
+		<p><label>Names:</label><input type='text' name='names' value='' class='auto'></p>
```
translate (use xlt and do not include the colon in the translation)

bradymiller commented on library/ajax/drug_autocomplete/index.php:
```diff
@@ -0,0 +1,28 @@
+<head>
+  <meta charset="utf-8">
+  <title>Demo</title>
+  <link rel="stylesheet" href="../../library/js/jquery-ui.min.css" type="text/css" /> 
```
remove lines 6, 14, 15 and then place following code here:
``````
<?php
html_header_show();
$include_standard_style_js = array("jquery-ui","jquery-ui-sunny");
require "{$GLOBALS['srcdir']}/templates/standard_header_template.php";
?>

bradymiller commented on library/ajax/drug_autocomplete/search.php:
```diff
@@ -0,0 +1,51 @@
+<?php
```
Remove the following files above:
library/ajax/drug_autocomplete/js/jquery-1.9.1.min.js
library/ajax/drug_autocomplete/js/jquery-ui.min.css
library/ajax/drug_autocomplete/js/jquery-ui.min.js
library/ajax/drug_autocomplete/js/jquery-ui.min.js

bradymiller commented on library/ajax/drug_autocomplete/search.php:
```diff
@@ -0,0 +1,51 @@
+ * @author  Sherwin Gaddis <sherwingaddis@gmail.com>
+ * @link    http://www.open-emr.org
+ */
+$fake_register_globals=false;
```
remove flag (as of recently, the codebase no longer uses these flags :) )

bradymiller commented on library/ajax/drug_autocomplete/search.php:
```diff
@@ -0,0 +1,51 @@
+ */
+$fake_register_globals=false;
+
+$sanitize_all_escapes=true;	
```
remove flag (as of recently, the codebase no longer uses these flags :) )

bradymiller commented on library/ajax/pharmacy_autocomplete/address.txt:
```diff
@@ -0,0 +1 @@
+SELECT id, Store_name, address_line_1, city, state FROM pharmacies_weno WHERE Store_name LIKE '%wal%' AND address_line_1 LIKE '%1%' AND city LIKE '%lubbock%'
```
What is this?

bradymiller commented on library/ajax/pharmacy_autocomplete/index.php:
```diff
@@ -0,0 +1,42 @@
+<?php 
```
Needs license/copyright header.

juggernautsei replied:
removed file

bradymiller commented on library/ajax/pharmacy_autocomplete/index.php:
```diff
@@ -0,0 +1,42 @@
+<head>
+  <meta charset="utf-8">
+  <title>Demo</title>
+  <link rel="stylesheet" href="../../js/jquery-ui.min.css" type="text/css" /> 
```
remove lines 12, 22, 23 and then place following code here:
``````
<?php
html_header_show();
$include_standard_style_js = array("jquery-ui","jquery-ui-sunny");
require "{$GLOBALS['srcdir']}/templates/standard_header_template.php";
?>

bradymiller commented on library/ajax/pharmacy_autocomplete/index.php:
```diff
@@ -0,0 +1,42 @@
+<body> 
+
+	<form action='' method='post'>
+		<p><label>Pharmacy Names : </label><input type='text' name='names' id="pharm" value='' class='auto'></p>
```
translate (use xlt)

juggernautsei replied:
removed file

bradymiller commented on library/ajax/pharmacy_autocomplete/index.php:
```diff
@@ -0,0 +1,42 @@
+
+	<form action='' method='post'>
+		<p><label>Pharmacy Names : </label><input type='text' name='names' id="pharm" value='' class='auto'></p>
+		<p><label>City : </label><input type='text' name='city' id="city" value='' placeholder='Enter City First'></p>
```
translate two items in above line:
``````
<p><label><?php echo xlt('City'); ?> : </label><input type='text' name='city' id="city" value='' placeholder='<?php xla('Enter City First'); ?>></p>

bradymiller commented on library/ajax/pharmacy_autocomplete/search.php:
```diff
@@ -0,0 +1,49 @@
+<?php
```
Remove the above files:
library/ajax/pharmacy_autocomplete/js/jquery-1.9.1.min.js
library/ajax/pharmacy_autocomplete/js/jquery-ui.min.css
library/ajax/pharmacy_autocomplete/js/jquery-ui.min.js

bradymiller commented on library/ajax/pharmacy_autocomplete/search.php:
```diff
@@ -0,0 +1,49 @@
+ * Sherwin Gaddis <sherwingaddis@gmail.com>
+ * @link    http://www.open-emr.org
+ */
+$fake_register_globals=false;
```
remove flag (as of recently, the codebase no longer uses these flags :) )

bradymiller commented on library/ajax/pharmacy_autocomplete/search.php:
```diff
@@ -0,0 +1,49 @@
+ * @link    http://www.open-emr.org
+ */
+$fake_register_globals=false;
+$sanitize_all_escapes=true;	
```
remove flag (as of recently, the codebase no longer uses these flags :) )

bradymiller commented on library/ajax/pharmacy_autocomplete/terms.txt:
```diff
@@ -0,0 +1 @@
+amarillo
```
what is this?

juggernautsei replied:
don't know, deleted!

bradymiller commented on library/classes/Prescription.class.php:
```diff
@@ -254,7 +257,10 @@ function set_unit($unit) {
+        //if (!empty($id) && is_numeric($id)) { $this->id = $id; }
+		$phid = explode("-", $id);
+		$id = $phid[0];
+        $this->id = $id;		
```
This code block seems odd. What is this code meant to do?

juggernautsei replied:
When there was a separate table for the pharmacies, to distinguish the two tables, there was a hyphen in the id. this was to remove the hyphen and get the id. No longer needed since we are going to use the native tables.

bradymiller commented on library/classes/Prescription.class.php:
```diff
@@ -479,11 +490,31 @@ function get_note() {
+        $sql = "SELECT NDC FROM erx_drug_paid WHERE drug_label_name LIKE ? ";
+        $val = array('%'.$drug.'%');
+        $getNDC = sqlStatement($sql, $val);
+		$ndc = sqlFetchArray($getNDC);
```
better to use following to replace above 2 lines:
``````
$ndc = sqlQuery($sql, $val);

bradymiller commented on sql/database.sql:
```diff
@@ -5384,6 +5384,8 @@ CREATE TABLE `pharmacies` (
+  `ncpdp` int(12) DEFAULT NULL,
+  `npi` int(12) DEFAULT NULL.
   PRIMARY KEY  (`id`)
 ) ENGINE=InnoDB;
```
Missing the new columns in the following tables:
users
prescriptions

juggernautsei replied:
I don't understand. I looked in the main repository and those columns are not there for me to copy.

bradymiller replied:
These are in your sql upgrade script.

bradymiller commented on sql/5_0_0-to-5_0_1_upgrade.sql:
```diff
@@ -129,6 +129,43 @@ VALUES ('page_validation','add_edit_event#theform_groups','/interface/main/calen
+-- 1. users
+--
+#IfMissingColumn users weno_prov_id
+ALTER TABLE `users` ADD `weno_prov_id` VARCHAR(15) NULL DEFAULT NULL;
```
"NULL DEFAULT NULL" seems odd

juggernautsei replied:
Got that straight from MySQL when it dumped the table structure.

bradymiller replied:
What you have is not wrong, but does not follow format of elements in database.sql.
Change it to "DEFAULT NULL"

bradymiller commented on sql/5_0_0-to-5_0_1_upgrade.sql:
```diff
@@ -129,6 +129,43 @@ VALUES ('page_validation','add_edit_event#theform_groups','/interface/main/calen
+-- 2. prescriptions 
+--
+#IfMissingColumn prescriptions ntx 
+ALTER TABLE `prescriptions` ADD `ntx` INT(2) NULL DEFAULT NULL;
```
"NULL DEFAULT NULL" seems odd

juggernautsei replied:
Got that straight from MySQL when it dumped the table structure.

bradymiller replied:
What you have is not wrong, but does not follow format of elements in database.sql.
Change it to "DEFAULT NULL"

bradymiller commented on sql/5_0_0-to-5_0_1_upgrade.sql:
```diff
@@ -129,6 +129,43 @@ VALUES ('page_validation','add_edit_event#theform_groups','/interface/main/calen
+-- 3. prescriptions 
+--
+#IfMissingColumn prescriptions ntx 
+ALTER TABLE `prescriptions` ADD `rtx` INT(2) NULL DEFAULT NULL;
```
"NULL DEFAULT NULL" seems odd

juggernautsei replied:
Got that straight from MySQL when it dumped the table structure.

bradymiller replied:
What you have is not wrong, but does not follow format of elements in database.sql.
Change it to "DEFAULT NULL"

juggernautsei replied:
This has been updated..

bradymiller commented on sql/5_0_0-to-5_0_1_upgrade.sql:
```diff
@@ -129,6 +129,43 @@ VALUES ('page_validation','add_edit_event#theform_groups','/interface/main/calen
+
+-- 3. prescriptions 
+--
+#IfMissingColumn prescriptions ntx 
```
the ntx should be a rtx

bradymiller commented on sql/5_0_0-to-5_0_1_upgrade.sql:
```diff
@@ -129,6 +129,43 @@ VALUES ('page_validation','add_edit_event#theform_groups','/interface/main/calen
+
+-- 4. prescriptions 
+--
+#IfMissingColumn prescriptions ntx 
```
the ntx should be txDate

bradymiller commented on sql/5_0_0-to-5_0_1_upgrade.sql:
```diff
@@ -129,6 +129,43 @@ VALUES ('page_validation','add_edit_event#theform_groups','/interface/main/calen
+
+-- 6. pharmacies 
+--
+#IfMissingColumn pharmacies ncpdp
```
the ncpdp should be npi

bradymiller commented on templates/prescription/general_edit.html:
```diff
@@ -149,90 +172,91 @@
+    <select name="drug_id" onchange="drugselected(this)">
     {html_options values=$DRUG_ARRAY_VALUES output=$DRUG_ARRAY_OUTPUT selected=$prescription->get_drug_id()}
-		</select>
-		<input type="hidden" name="rxnorm_drugcode" value="{$prescription->rxnorm_drugcode}">
```
Is this being remove for a reason?

bradymiller replied:
disregard above question

bradymiller commented on templates/prescription/general_edit.html:
```diff
@@ -4,21 +4,42 @@
 
 <link rel="stylesheet" href="{$CSS_HEADER}" type="text/css">
```
PLACEMARKER FOR ONGOING REVIEW

bradymiller commented:
@juggernautsei ,
Just let me know when your changes are complete. Then I'm happy to rebase this into 1 commit and place on most recent codebase (and I'll then give you instructions on how to bring my branch into your repository). Looking very forward to getting this feature into the codebase. -brady

juggernautsei commented:
@bradymiller 
I thought I was finished with all of the requested revisions. I carefully went through from the beginning and made sure I commented on each of the items you listed and removed files that were not needed and updated all requests. You can go ahead with the rebase and the send me the instructions.

bradymiller commented:
sounds good. will try to get the rebased commit/branch out by tomorrow :)

bradymiller commented:
Rebased all you code into 1 commit on top of current codebase here. Here are command line instructions to get it:
``````
#add a remote link to my github repo
git remote add bradymiller https://github.com/bradymiller/openemr.git
#collect my repo
git fetch bradymiller
#checkout my branch
git checkout bradymiller/new_erx_2
#now create your own branch from mine
git checkout -b new_erx_2
#now push your branch to your github repo
git push origin new_erx_2
#after do above, then create a new Pull Request on github with you new branch
``````
Once you have placed above new branch on a new PR, then will do another review.

juggernautsei commented:
https://github.com/openemr/openemr/pull/829
